tests/core/action_test.py

```diff
@@ -96,3 +102,116 @@ def test_from_config_none_values(self):
         assert command_config.secret_env == {}
         assert command_config.secret_volumes == []
         assert command_config.extra_volumes == set()
+
+    def action_command_config_json(self):
```

Comment:
for example if this key would to be raw json: ``"extra_volumes": [serialize_namedtuple(volume) for volume in extra_volumes],`` we wont be testing converting from the ``ConfigVolume`` class and the call to the serialize_namedtuple function

Reply:
gotcha, that said: i think we can cover the serialization in different tests and focus on deserialization for the tests you're adding here - in which case, we don't need to worry about trying to be consistent with how we serialize things since we're focusing on going from a json string back into a python object :)